scripts/sct_register_to_template.py

```diff
@@ -72,133 +75,210 @@ def __init__(self):
-                      mandatory=False)
-    parser.add_option(name='-centerline-algo',
-                      type_value='multiple_choice',
-                      description='Algorithm for centerline fitting (when straightening the spinal cord).',
-                      mandatory=False,
-                      example=['polyfit', 'bspline', 'linear', 'nurbs'],
-                      default_value=ParamCenterline().algo_fitting)
-    parser.add_option(name='-centerline-smooth',
-                      type_value='int',
-                      description='Degree of smoothing for centerline fitting. Only use with -centerline-algo {bspline, linear}.',
-                      mandatory=False,
-                      default_value=ParamCenterline().smooth)
-    parser.add_option(name='-qc',
-                      type_value='folder_creation',
-                      description='The path where the quality control generated content will be saved',
-                      default_value=param.path_qc)
-    parser.add_option(name='-qc-dataset',
-                      type_value='str',
-                      description='If provided, this string will be mentioned in the QC report as the dataset the process was run on',
-                      )
-    parser.add_option(name='-qc-subject',
-                      type_value='str',
-                      description='If provided, this string will be mentioned in the QC report as the subject the process was run on',
-                      )
-    parser.add_option(name="-igt",
-                      type_value="image_nifti",
-                      description="File name of ground-truth template cord segmentation (binary nifti).",
-                      mandatory=False)
-    parser.add_option(name="-r",
-                      type_value="multiple_choice",
-                      description="""Remove temporary files.""",
-                      mandatory=False,
-                      default_value=param.remove_temp_files,
-                      example=['0', '1'])
-    parser.add_option(name="-v",
-                      type_value="multiple_choice",
-                      description="""Verbose. 0: nothing. 1: basic. 2: extended.""",
-                      mandatory=False,
-                      default_value=param.verbose,
-                      example=['0', '1', '2'])
+    parser = argparse.ArgumentParser(
+        description=(
+            "Register an anatomical image to the spinal cord MRI template (default: PAM50).\n"
+            "\n"
+            "The registration process includes three main registration steps:\n"
+            "  1. straightening of the image using the spinal cord segmentation (see sct_straighten_spinalcord for "
+            "details);\n"
```

Comment:
again a comment about indentation for list. Currently doing that:
~~~
The registration process includes three main registration steps:
  1. straightening of the image using the spinal cord segmentation (see sct_straighten_spinalcord fo
r details);
  2. vertebral alignment between the image and the template, using labels along the spine;
  3. iterative slice-wise non-linear registration (see sct_register_multimodal for details)
~~~
where this would be much nicer:
~~~
The registration process includes three main registration steps:
  1. straightening of the image using the spinal cord segmentation (see sct_straighten_spinalcord fo
     r details);
  2. vertebral alignment between the image and the template, using labels along the spine;
  3. iterative slice-wise non-linear registration (see sct_register_multimodal for details)
~~~